### PR TITLE
[Dynamo] Fix TIMM benchmark compute_loss

### LINKED ISSUE #97382: amp + eager backend + training failing for some timm models
### 🐛 Describe the bug

```
python benchmarks/dynamo/timm_models.py --backend eager --only mnasnet_100 --accuracy --amp --training
```
Results in 
```
[2023-03-22 21:57:03,055] torch._dynamo.utils: [ERROR] RMSE (res-fp64): 0.09759, (ref-fp64): 0.01743 and shape=torch.Size([8, 1000])
```
for me.

interestingly, switching to the `aot_eager` backend allows it to pass.

```
python benchmarks/dynamo/timm_models.py --backend eager --only mnasnet_100 --accuracy --amp --training
```

The cause of this breakage is https://github.com/pytorch/pytorch/pull/95416.

cc: @yanboliang @davidberard98

### Versions

N/A

cc @ezyang @soumith @msaroufim @wconstab @ngimel @bdhirsh @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @desertfire

### PR DESCRIPTION
**Pull request by yanboliang**
Fixes #97382

#95416 fixed a critical bug in dynamo benchmark, where AMP tests fall back to eager mode before that PR. However, after that PR, we found [a list of TIMM models amp + eager + training testing failed](https://docs.google.com/spreadsheets/d/1DEhirVOkj15Lu4UNawIUon9MqkVLaWqyT-DQPif5NHk/edit#gid=0). 
Now we identified the root cause is: high loss values make gradient checking harder, as small changes in accumulation order upset accuracy checks. We should switch to the helper function ```reduce_to_scalar_loss``` which has been used by Torchbench tests.
After switching to ```reduce_to_scalar_loss```, TIMM models accuracy pass rate grows from 67.74% to 91.94% in my local test. The rest 5 failed models(ese_vovnet19b_dw, fbnetc_100, mnasnet_100, mobilevit_s, sebotnet33ts_256) need further investigation and handling, but I think it should be similar reason.

cc @soumith @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @desertfire

### CODE REVIEW

**Review by Chillee — APPROVED** [review]
:O 

Perhaps I just got unlucky 🤔 I was looking at mnasnet, and I noticed that both the loss *and* the output are quite a bit less accurate.
